tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tracing.opentelemetry;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.api.baggage.BaggageEntry;
+import io.opentelemetry.api.baggage.BaggageEntryMetadata;
+
+class MutableOpenTelemetryBaggage implements Baggage {
+
+    private final Map<String, BaggageEntry> values = new LinkedHashMap<>();
+
+    MutableOpenTelemetryBaggage() {
+    }
+
+    private MutableOpenTelemetryBaggage(Builder builder) {
+        values.putAll(builder.values);
+    }
+
+    @Override
+    public int size() {
+        return values.size();
+    }
+
+    @Override
+    public void forEach(BiConsumer<? super String, ? super BaggageEntry> consumer) {
+        values.forEach(consumer);
+    }
+
+    @Override
+    public Map<String, BaggageEntry> asMap() {
+        return Collections.unmodifiableMap(values);
```

Comment:
Depending on how often this method is called, you may want to store the unmodifiable view as a `final` instance field.

Reply:
In the world of the neutral Helidon tracing API, the contents of the baggage data structure can change even once the span is started. So `values` can change at any time.

Reply:
What I mean is:
```
// in your instance field declarations
private Map<String, BaggageEntry> values; // as before; don't have it in front of me
private Map<String, BaggageEntry> valuesView;

// In your class' constructor:
this.values = new HashMap<>(); // or whatever; all mutation methods target this
this.valuesView = Collections.unmodifiableMap(this.values); // return this from getters
```

Reply:
Ah, yes. Got it. Will change. Was there a reason you didn't advocate for this:

```java
private final Map<String, BaggageEntry> values = new HashMap<>();
private final Map<String, BaggageEntry> valuesView = Collections.unmodifiableMap(values);
```

Reply:
Not particularly; I tend to do all my instance field initializations in the constructor. Style and habit I suppose.

Reply:
Changed to save the view.